common/retry/retry.go

```diff
@@ -1,30 +1,19 @@
 package retry
 
-import "time"
+type RetryFunc func() (shouldRetry bool)
 
-type RetryFunc func() (shouldRetry bool, err error)
-
-func Retry(fn RetryFunc, options ...Option) error {
+func Retry(shouldRetryFN RetryFunc, options ...Option) error {
```

Comment:
I'm wondering if this should still return an error, since removing it would make it so the option errors are swallowed, the other option would be to move the options to a single struct.